Rust/src/lib.rs

```diff
@@ -1,31 +1,30 @@
-mod boundingbox;
-mod cut;
-mod imputevisitor;
-mod intervalstoremanager;
-pub mod multidimdatawithkey;
-mod nodestore;
-mod nodeview;
+pub mod common;
 mod pointstore;
-mod randomcuttree;
 pub mod rcf;
-mod sampler;
 mod samplerplustree;
-pub mod samplesummary;
-mod scalarscorevisitor;
 mod types;
 mod visitor;
-mod conditionalfieldsummarizer;
+
 extern crate rand;
+extern crate rand_chacha;
 
 use num::abs;
-use crate::rcf::{create_rcf, RCF};
-extern crate rand_chacha;
 
-pub fn L1distance(a: &[f32], b : &[f32]) -> f64{
+pub fn vec_l1distance(a: &Vec<f32>, b : &[f32]) -> f64 {
+    l1distance(a, b)
```

Comment:
You should be able to write a single generic version of this function. Here's a version that that takes two iterators that iterate over `&f32`s. There should be a way to write it so that we only require that the types being iterated over can be cast to f64, but I was having trouble getting the type bounds right for that.

```rust
use num::abs;

fn l1distance<'a, L, R>(left: L, right: R) -> f64
where
  L: IntoIterator<Item=&'a f32>,
  R: IntoIterator<Item=&'a f32>,
{
    left.into_iter().zip(right).map(|(a, b)| abs(f64::from(*a) - f64::from(*b))).sum()
}

fn main() {
    let point1: Vec<f32> = vec![1.0, 3.0, 5.0];
    let point2: Vec<f32> = vec![2.0, -1.0, 7.0];
    let point2_as_slice: &[f32] = &point2[..];
    
    println!("|point1 - point2| = {}", l1distance(&point1, &point2));
    println!("|point1 - point2_as_slice| = {}", l1distance(&point1, point2_as_slice));
}
```

Reply:
This seemed to be needed due to the paramteric_test macro -- it is possible something is amiss there. But changing to IntoIterator is forcing `for<'r, 's> fn(&'r [f32], &'s [f32]) ->  .... and then recursively it is forcing all the function call definitions to split. It would be a bad idea to get into annotations across the different calls, we would not be able to handle recursion. If this is galling then we should unroll the parametric test. We can also move this definition from the lib to the test.

Reply:
Fixed with changes